packages/next-swc/crates/next-core/src/app_structure.rs

```diff
@@ -473,7 +473,27 @@ pub enum Entrypoint {
 }
 
 #[turbo_tasks::value(transparent)]
-pub struct Entrypoints(IndexMap<String, Entrypoint>);
+pub struct Entrypoints(IndexMap<AppPath, Entrypoint>);
+
+#[turbo_tasks::value_impl]
+impl Entrypoints {
+    #[turbo_tasks::function]
+    pub fn paths(&self) -> Vc<EntrypointPaths> {
+        Vc::cell(self.0.keys().cloned().collect())
+    }
+}
+
+#[turbo_tasks::value(transparent)]
+#[derive(Default)]
+pub struct EntrypointPaths(Vec<AppPath>);
```

Comment:
Do we need this wrapper struct? Couldn't it just be return `Vc<Vec<AppPath>>` directly?

Reply:
I think that was only possible if it's `Vc<Vec<Vc<AppPath>>>>`